custom_components/bmw_connected_drive/device_tracker.py

```diff
@@ -86,8 +96,8 @@ def force_update(self):
     def update(self):
         """Update state of the decvice tracker."""
```

Comment:
You should update `self._heading` in this function as well. Because now you only update heading once during init and then it gets never updated anymore.